### PR TITLE
[CT-1841] Convert custom target test to Pytest

### LINKED ISSUE #6638: Convert 075_custom_target_path tests

### PR DESCRIPTION
**Pull request by aranke**
resolves #6638 

### Checklist

- [x] I have read [the contributing guide](https://github.com/dbt-labs/dbt-core/blob/main/CONTRIBUTING.md) and understand what's expected of me
- [x] I have signed the [CLA](https://docs.getdbt.com/docs/contributor-license-agreements)
- [x] I have run this code in development and it appears to resolve the stated issue
- [x] This PR includes tests, or tests are not required/relevant for this PR
- [x] I have [opened an issue to add/update docs](https://github.com/dbt-labs/docs.getdbt.com/issues/new/choose), or docs changes are not required/relevant for this PR
- [x] I have run `changie new` to [create a changelog entry](https://github.com/dbt-labs/dbt-core/blob/main/CONTRIBUTING.md#adding-a-changelog-entry)

### CODE REVIEW

**Review by ChenyuLInx — APPROVED** [review]
Looks

**Review by MichelleArk — APPROVED** [review]

**Review by ChenyuLInx — APPROVED** [review]

File: tests/functional/custom_target_path/test_custom_target_path.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+import pytest
+
+from dbt.tests.util import run_dbt
+
+
+class TestTargetPathConfig:
+    @pytest.fixture(scope="class")
+    def project_config_update(self):
+        return {"config-version": 2, "target-path": "project_target"}
```
Comment (VersusFacit) [inline tests/functional/custom_target_path/test_custom_target_path.py:11]: nit: we don't need the `"config-version": 2`

File: tox.ini [hunk 2]
```diff
@@ -25,7 +25,6 @@ passenv =
   POSTGRES_TEST_*
   PYTEST_ADDOPTS
 commands =
-  {envpython} -m pytest --cov=core -m profile_postgres {posargs} test/integration
```
Comment (VersusFacit) [inline tox.ini:28]: ❤️

**Review by VersusFacit — APPROVED** [review]
Looks great overall! So excited to see this complete.

File: tox.ini (hunk 2 above)
Comment (jtcohen6) [reply tox.ini:28]: this is so, so satisfying

**Pull request merged**